src/primitive_esp32.cc

```diff
@@ -521,6 +529,51 @@ PRIMITIVE(memory_page_report) {
   return result;
 }
 
+PRIMITIVE(watchdog_init) {
+  ARGS(uint32, ms);
+
+  int watchdog = watchdog_timers.any();
+  if (watchdog == kInvalidWatchdogTimer) FAIL(OUT_OF_RANGE);
+
+  esp_task_wdt_config_t config = {
+    .timeout_ms = ms,
+    .idle_core_mask = 0,
+    .trigger_panic = true,
+  };
+  esp_err_t err = esp_task_wdt_init(&config);
+  if (err != ESP_OK) {
+    return Primitive::os_error(err, process);
+  }
+
+  SystemEventSource::instance()->run([&]() {
+    err = esp_task_wdt_add(null);  // Add the current thread.
```

Comment:
I feel 'current' is overloaded here.

```suggestion
    err = esp_task_wdt_add(null);  // Add the SystemEventSource thread.
```

Reply:
done.